Objects/listobject.c

```diff
@@ -656,6 +656,18 @@ list_item(PyObject *aa, Py_ssize_t i)
         return NULL;
     }
     PyObject *item;
+#ifdef Py_GIL_DISABLED
```

Comment:
I think the fast path is already implemented in [list_get_item_ref](https://github.com/python/cpython/blob/65de194dd80bbc8cb7098d21cfd6aefd11d0d0ce/Objects/listobject.c#L360-L385), so this can just be something like

```
return list_get_item_ref(a, i);
```

I think we also need to set the `_Py_REF_MAYBE_WEAKREF` bit in the shared part of the refcount in the slow path in `list_item_impl`. This is needed to ensure that future calls to `_Py_TryXGetRef` in the fast path succeed when the returned item is not owned by the calling thread. This could be something like replacing [the call to](https://github.com/python/cpython/blob/65de194dd80bbc8cb7098d21cfd6aefd11d0d0ce/Objects/listobject.c#L354) `Py_NewRef` in `list_item_impl` with a call to [`_Py_NewRefWithLock`](https://github.com/python/cpython/blob/65de194dd80bbc8cb7098d21cfd6aefd11d0d0ce/Include/internal/pycore_object.h#L495).